### DISCUSSION TITLE
Hotfix Salt Issue?

### DISCUSSION THREAD

**Discussion by argwfm** [Security-Onion-Solutions/securityonion#7773] [Unsupported Versions]
Applied the latest 2.3.110 hotfix 20220407 (skipped 20220401/05), and everything went w/out issue on the master. Minions are having issues. Running `salt-call state.highstate` only returns `Succeeded: 31 (changed=3)`, due to an apparent issue updating salt w/ the below `bootstrap-salt.sh` command. Rebooting before manual work-around results in `so-status` returning wait_start for all.

Manual work-around was to remove the `proxy=` declaration from `/etc/yum.conf` and replace `/etc/yum.repos.d/securityonioncache.repo` w/ the version from the manager on every node/minion. After this this manually executing `bootstrap-salt.sh` completes w/out error and `salt-call state.highstate` returns as expected.

Is this a known issue? Is there a way to force a rebuild/update to the master's repocache so this doesn't fail across the grid?

```
/usr/sbin/bootstrap-salt.sh -s 120 -r -F -x python3 stable 3004.1
 *  INFO: Running version: 2021.09.17
 *  INFO: Executed by: -
 *  INFO: Command line: '/usr/sbin/bootstrap-salt.sh -s 120 -r -F -x python3 stable 3004.1'

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-1160.59.1.el7.x86_64
 *  INFO:   Distribution: CentOS 7.9

 *  INFO: Installing minion
 *  INFO: Found function install_centos_stable_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_centos_stable
 *  INFO: Found function install_centos_stable_post
 *  INFO: Found function install_centos_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_centos_check_services
 *  INFO: Running install_centos_stable_deps()
 *  WARN: Detected -r or -R option while installing Salt packages for Python 3.
 *  WARN: Python 3 packages for older Salt releases requires the EPEL repository to be installed.
 *  WARN: Installing the EPEL repository automatically is disabled when using the -r or -R options.
Loaded plugins: versionlock
http://repocache.securityonion.net/file/securityonion-repo/base/repodata/repomd.xml: [Errno 14] HTTP Error 500 - Internal Server Error
Trying other mirror.

 One of the configured repositories failed (CentOS-7 - Base),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this: [...]

failure: repodata/repomd.xml from base: [Errno 256] No more mirrors to try.
http://repocache.securityonion.net/file/securityonion-repo/base/repodata/repomd.xml: [Errno 14] HTTP Error 500 - Internal Server Error
 * ERROR: Failed to run install_centos_stable_deps()!!!
```

**Comment by dougburks** [comment]
Have you tried restarting the so-aptcacherng service on the manager?

**Reply by argwfm** [reply]
Thanks for the reply- yes, I restarted the manager after confirming state.highstate completed (grid sat for hours before I began troubleshooting); restart/recycling containers had no effect, only fix I found as noted above.

**Reply by dougburks** [reply]
Are there any network firewalls between the nodes?

Are the nodes able to connect to port 3142 on the manager?

**Reply by argwfm** [reply]
Yes, 3142 is accessible... logs/PCAP are showing the below. I've verified `/opt/so/conf/aptcacher-ng/etc/acng.conf` matches the SO default, but I can't find where it's pulling an SSL config from. Appreciate the support!

```
failure: repodata/repomd.xml from base: [Errno 256] No more mirrors to try.
http://repocache.securityonion.net/file/securityonion-repo/base/repodata/repomd.xml: [Errno 14] HTTP Error 500 - Internal Server Error
```
```
Server: Debian Apt-Cacher NG/3.3.1
<!DOCTYPE html>
<html lang="en"><head><title>500 SSL error: Generic SSL failure</title>
</head>
<body><h1>500 SSL error: Generic SSL failure</h1></body></html>
```

**Reply by dougburks** [reply]
You could try removing the aptcacher-ng cache.  From the manager:
```
# stop the ACNG container
sudo docker stop so-aptcacherng

# remove the ACNG container
sudo docker rm so-aptcacherng

# remove the ACNG cache directory
sudo rm -rf /opt/so/conf/aptcacher-ng/cache/

# apply the ACNG state
sudo salt-call state.apply manager
```

**Reply by argwfm** [reply]
Will this reinstall it, or permanently remove?

Disabling via _managerupdate_ resolves, but would like to fix aptcacher-ng, if possible.
```
/opt/so/saltstack/local/pillars
managerupdate: 0
```

**Reply by dougburks** [reply]
If you run the commands above, then aptcacher-ng should be back up and running, just with an empty cache.

**Reply by argwfm** [reply]
No go w/ removing/replacing the so-aptcacherng docker image, same SSL errors.

```
failure: repodata/repomd.xml from base: [Errno 256] No more mirrors to try.
http://repocache.securityonion.net/file/securityonion-repo/base/repodata/repomd.xml: [Errno 14] HTTP Error 500 - Internal Server Error
```

```
Server: Debian Apt-Cacher NG/3.3.1
<!DOCTYPE html>
<html lang="en"><head><title>500 SSL error: Generic SSL failure</title>
</head>
<body><h1>500 SSL error: Generic SSL failure</h1></body></html>
```

**Reply by dougburks** [reply]
Have you checked `/opt/so/log/aptcacher-ng/` for additional clues?

**Reply by argwfm** [reply]
Yes, same generic error received by yum on the minions-
`|securityonion/file/securityonion-repo/base/repodata/repomd.xml [HTTP error, code: 500]`

**Reply by dougburks** [reply]
Have you double-checked that your manager can connect to repo.securityonion.net?

**Reply by argwfm** [reply]
Indeed, no issues there... repo.securityonion.net is reachable from the manager.
